src/VDPCommandRegViewer.cpp

```diff
@@ -193,34 +193,33 @@ void VDPCommandRegViewer::on_syncPushButton_clicked()
 
 void VDPCommandRegViewer::on_launchPushButton_clicked()
 {
-	unsigned char newregs[64];
-	memset(newregs, 0, 64);
-	newregs[32] = stringToValue(lineEdit_r32->text());
-	newregs[33] = stringToValue(lineEdit_r33->text());
-	newregs[34] = stringToValue(lineEdit_r34->text());
-	newregs[35] = stringToValue(lineEdit_r35->text());
-	newregs[36] = stringToValue(lineEdit_r36->text());
-	newregs[37] = stringToValue(lineEdit_r37->text());
-	newregs[38] = stringToValue(lineEdit_r38->text());
-	newregs[39] = stringToValue(lineEdit_r39->text());
-	newregs[40] = stringToValue(lineEdit_r40->text());
-	newregs[41] = stringToValue(lineEdit_r41->text());
-	newregs[42] = stringToValue(lineEdit_r42->text());
-	newregs[43] = stringToValue(lineEdit_r43->text());
-	newregs[44] = stringToValue(lineEdit_r44->text());
-	newregs[45] = stringToValue(lineEdit_r45->text());
-	newregs[46] = stringToValue(lineEdit_r46->text());
+	uint8_t newregs[64] = {};
+	newregs[32] = stringToValue<uint8_t>(lineEdit_r32->text()).value_or(-1);
+	newregs[33] = stringToValue<uint8_t>(lineEdit_r33->text()).value_or(-1);
+	newregs[34] = stringToValue<uint8_t>(lineEdit_r34->text()).value_or(-1);
+	newregs[35] = stringToValue<uint8_t>(lineEdit_r35->text()).value_or(-1);
+	newregs[36] = stringToValue<uint8_t>(lineEdit_r36->text()).value_or(-1);
+	newregs[37] = stringToValue<uint8_t>(lineEdit_r37->text()).value_or(-1);
+	newregs[38] = stringToValue<uint8_t>(lineEdit_r38->text()).value_or(-1);
+	newregs[39] = stringToValue<uint8_t>(lineEdit_r39->text()).value_or(-1);
+	newregs[40] = stringToValue<uint8_t>(lineEdit_r40->text()).value_or(-1);
+	newregs[41] = stringToValue<uint8_t>(lineEdit_r41->text()).value_or(-1);
+	newregs[42] = stringToValue<uint8_t>(lineEdit_r42->text()).value_or(-1);
+	newregs[43] = stringToValue<uint8_t>(lineEdit_r43->text()).value_or(-1);
+	newregs[44] = stringToValue<uint8_t>(lineEdit_r44->text()).value_or(-1);
+	newregs[45] = stringToValue<uint8_t>(lineEdit_r45->text()).value_or(-1);
+	newregs[46] = stringToValue<uint8_t>(lineEdit_r46->text()).value_or(-1);
```

Comment:
Yes, I though it was confusing too, but I kept it as it is because that is what the code did after all. Didn't want to change something that deserves its own issue, but OK. I will change it.